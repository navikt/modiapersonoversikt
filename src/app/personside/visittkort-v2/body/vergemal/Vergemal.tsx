import Hjelpetekst from 'nav-frontend-hjelpetekst';
import { Element, Feilmelding, Normaltekst } from 'nav-frontend-typografi';
import { Fragment } from 'react';
import styled from 'styled-components';
import EtikettGraa from '../../../../../components/EtikettGraa';
import VergemalLogo from '../../../../../svg/Utropstegn';
import { InformasjonElement, type Verge as VergeInterface } from '../../PersondataDomain';
import { harFeilendeSystemer } from '../../harFeilendeSystemer';
import { hentNavn } from '../../visittkort-utils';
import GyldighetsPeriode from '../GyldighetsPeriode';
import VisittkortElement from '../VisittkortElement';
import { VisittkortGruppe } from '../VisittkortStyles';

const Vergeinformasjon = styled.div`
    margin-bottom: 5px;
`;

interface Props {
    feilendeSystemer: Array<InformasjonElement>;
    vergemal: VergeInterface[];
}

function Verge(props: { feilendeSystemer: Array<InformasjonElement>; verge: VergeInterface }) {
    const { verge } = props;
    const harFeilendeSystemOgIngenNavn =
        harFeilendeSystemer(props.feilendeSystemer, InformasjonElement.PDL_TREDJEPARTSPERSONER) && !verge.navn ? (
            <Feilmelding>Feilet ved uthenting av navn p책 verge</Feilmelding>
        ) : (
            <Normaltekst>{hentNavn(verge.navn, 'Navn ikke tilgjengelig')}</Normaltekst>
        );

    return (
        <VisittkortElement beskrivelse="Verge">
            <Vergeinformasjon>
                {harFeilendeSystemOgIngenNavn}
                <Normaltekst>{verge.ident}</Normaltekst>
            </Vergeinformasjon>
            {verge.tjenesteOppgaver?.length > 0 ? (
                <Fragment>
                    <Element>
                        Omr책de{' '}
                        <Hjelpetekst>
                            Viser omr책dene verge har innsynsrett i.
                            <br />
                            Viser kun tjenesteoppgaver knyttet til NAV.
                        </Hjelpetekst>
                    </Element>
                    <Normaltekst>{verge.tjenesteOppgaver.join(', ')}</Normaltekst>
                </Fragment>
            ) : (
                <Fragment>
                    <Element>Omfang</Element>
                    <Normaltekst>{verge.omfang}</Normaltekst>
                </Fragment>
            )}
            <EtikettGraa>{verge.embete}</EtikettGraa>
            <GyldighetsPeriode gyldighetsPeriode={verge.gyldighetsPeriode} />
        </VisittkortElement>
    );
}

function Vergesakstype(props: { vergemal: VergeInterface[] }) {
    const alleVergesakstyper = props.vergemal.map((verge) => verge.vergesakstype);
    const unikeVergessakstyper = Array.from(new Set(alleVergesakstyper)).join(', ');
    return <Normaltekst>Vergesakstyper: {unikeVergessakstyper}</Normaltekst>;
}

function Vergemal({ feilendeSystemer, vergemal }: Props) {
    if (vergemal.isEmpty()) {
        return null;
    }

    return (
        <VisittkortGruppe ikon={<VergemalLogo />} tittel="Bruker er under vergem책l">
            <Vergesakstype vergemal={vergemal} />
            {vergemal.map((verge, index) => (
                <Verge feilendeSystemer={feilendeSystemer} verge={verge} key={index} />
            ))}
        </VisittkortGruppe>
    );
}

export default Vergemal;
