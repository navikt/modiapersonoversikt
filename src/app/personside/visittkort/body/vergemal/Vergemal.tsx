import * as React from 'react';
import styled from 'styled-components';

import EtikettLiten from 'nav-frontend-typografi/lib/etikett-liten';
import Undertekst from 'nav-frontend-typografi/lib/undertekst';

import VisittkortElement, { TittelStyle } from '../VisittkortElement';
import { Periode, Verge, Vergemal } from '../../../../../models/vergemal/vergemal';
import { formaterDato } from '../../../../../utils/dateUtils';
import Vergem책lLogo from '../../../../../svg/Utropstegn';
import EtikettMini from '../../../../../components/EtikettMini';
import { ENDASH } from '../../../../../utils/string-utils';

export const feilmelding = 'Feil ved visning av vergem책l';

function Periode(props: {periode: Periode}) {
    const {periode} = props;

    const fom = periode.fom ? formaterDato(periode.fom) : '';
    const tom = periode.tom ? formaterDato(periode.tom) : '';

    return(
        <EtikettMini>{fom} {ENDASH} {tom}</EtikettMini>
    );
}

const Vergeinformasjon = styled.div`
  padding-bottom: .5em;
`;

const VergeDiv = styled.div`
  margin-top: 1em;
  padding-bottom: 1.5em;
`;

const Vergesakstype = styled.span`
  display: block;
`;

function Verge(props: {verge: Verge}) {
    const {verge} = props;
    return (
        <VergeDiv>
            <EtikettLiten><TittelStyle>Verge</TittelStyle></EtikettLiten>
            <Vergeinformasjon>
                <Undertekst>{verge.navn.sammensatt}</Undertekst>
                <Undertekst>{verge.ident}</Undertekst>
                <Undertekst>{verge.vergetype ? verge.vergetype.beskrivelse : ''}</Undertekst>
            </Vergeinformasjon>
            <EtikettLiten><TittelStyle>Mandat</TittelStyle></EtikettLiten>
            <Undertekst>{verge.mandattype ? verge.mandattype.beskrivelse : ''}</Undertekst>
            <Undertekst>{verge.mandattekst || ''}</Undertekst>
            <EtikettMini>{verge.embete ? verge.embete.beskrivelse : ''}</EtikettMini>
            <Periode periode={verge.virkningsperiode}/>
        </VergeDiv>
    );
}

function Vergemal(props: {vergemal: Vergemal}) {
    const alleVergesakstyper = props.vergemal.verger.map(verge => verge.vergesakstype ?
        verge.vergesakstype.beskrivelse : 'Ingen vergesakstype oppgitt').join(', ');
    const verger = props.vergemal.verger.map(verge =>
        <Verge verge={verge} key={verge.ident}/>);
    return (
        <VisittkortElement
            beskrivelse={'Bruker er under vergem책l'}
            ikon={<Vergem책lLogo />}
            type={'header'}
        >
            <>
                <Vergesakstype>
                    <Undertekst>{alleVergesakstyper}</Undertekst>
                </Vergesakstype>
                {verger}
            </>
        </VisittkortElement>
    );
}

function VergemalWrapper(props: {vergemal?: Vergemal}) {
    if (!props.vergemal) {
        return <p>{feilmelding}</p>;
    }

    if (props.vergemal.verger.length === 0) {
        return null;
    }

    return (
        <Vergemal vergemal={props.vergemal}/>
    );

}

export default VergemalWrapper;