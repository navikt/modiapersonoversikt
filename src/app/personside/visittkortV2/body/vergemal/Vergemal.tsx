import * as React from 'react';

import { ENDASH, formaterDato } from '../../../../../utils/string-utils';
import styled from 'styled-components/macro';
import { Verge as VergeInterface } from '../../../../../models/personPdl/verge';
import VisittkortElement from '../VisittkortElement';
import { Element, Normaltekst } from 'nav-frontend-typografi';
import { vergeOmfangMapper } from './VergemalMapper';
import EtikettGraa from '../../../../../components/EtikettGraa';
import { VisittkortGruppe } from '../VisittkortStyles';
import VergemalLogo from '../../../../../svg/Utropstegn';
import Vergesakstype from './Vergesakstype';

function getPeriodeTekst(gyldighetstidspunkt?: string, opphoerstidspunkt?: string | null) {
    const fom = gyldighetstidspunkt ? formaterDato(gyldighetstidspunkt) : '';
    const tom = opphoerstidspunkt ? formaterDato(opphoerstidspunkt) : '';

    return `${fom} ${ENDASH} ${tom}`;
}

const Vergeinformasjon = styled.div`
    margin-bottom: 5px;
`;

function Verge(props: { verge: VergeInterface }) {
    const { verge } = props;

    const gyldighetstidspunkt = verge.gyldighetstidspunkt ? formaterDato(verge.gyldighetstidspunkt) : undefined;
    const opphoerstidspunkt = verge.opphoerstidspunkt ? formaterDato(verge.opphoerstidspunkt) : undefined;

    return (
        <VisittkortElement beskrivelse={'Verge'}>
            <Vergeinformasjon>
                <Normaltekst>
                    {verge.navn ? `${verge.navn.fornavn} ${verge.navn.etternavn}` : 'Navn ikke tilgjengelig'}
                </Normaltekst>
                <Normaltekst>{verge.ident || ''}</Normaltekst>
            </Vergeinformasjon>

            <Element>Omfang</Element>
            <Normaltekst>{vergeOmfangMapper[verge.omfang ?? 'undefined'] ?? verge.omfang}</Normaltekst>
            <EtikettGraa>
                {verge.embete ? verge.embete : ''}
                {verge.embete ? <br /> : ''}
                {getPeriodeTekst(gyldighetstidspunkt, opphoerstidspunkt)}
            </EtikettGraa>
        </VisittkortElement>
    );
}

function Vergemal(props: { vergemal: VergeInterface[] }) {
    const verger = props.vergemal.map((verge, index) => <Verge verge={verge} key={index} />);
    return (
        <VisittkortGruppe ikon={<VergemalLogo />} tittel="Bruker er under vergemÃ¥l">
            <Vergesakstype verger={props.vergemal} />
            {verger}
        </VisittkortGruppe>
    );
}

function VergemalWrapper(props: { vergemal: VergeInterface[] }) {
    if (props.vergemal.length === 0) {
        return null;
    }

    return <Vergemal vergemal={props.vergemal} />;
}

export default VergemalWrapper;
