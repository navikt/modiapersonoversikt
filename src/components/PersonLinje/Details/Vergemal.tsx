import { ExclamationmarkTriangleFillIcon } from '@navikt/aksel-icons';
import { Alert, BodyShort, Box, Detail, HStack, HelpText } from '@navikt/ds-react';
import { usePersonData } from 'src/lib/clients/modiapersonoversikt-api';
import { type PersonData, PersonDataFeilendeSystemer } from 'src/lib/types/modiapersonoversikt-api';
import ValidPeriod from '../common/ValidPeriod';
import { harFeilendeSystemer, hentNavn } from '../utils';
import { Group, InfoElement } from './components';

type Verge = PersonData['vergemal'][0];

function Verge(props: {
    feilendeSystemer: PersonDataFeilendeSystemer[];
    verge: Verge;
}) {
    const { verge } = props;
    const harFeilendeSystemOgIngenNavn =
        harFeilendeSystemer(props.feilendeSystemer, PersonDataFeilendeSystemer.PDL_TREDJEPARTSPERSONER) &&
        !verge.navn ? (
            <Alert variant="warning">Feilet ved uthenting av navn på verge</Alert>
        ) : (
            <BodyShort size="small">{hentNavn(verge.navn, 'Navn ikke tilgjengelig')}</BodyShort>
        );

    return (
        <InfoElement title="Verge">
            <Box className="mb-2">
                {harFeilendeSystemOgIngenNavn}
                <Detail>{verge.ident}</Detail>
            </Box>
            {verge.tjenesteOppgaver && verge.tjenesteOppgaver?.length > 0 ? (
                <>
                    <HStack>
                        <BodyShort weight="semibold" size="small">
                            Område{' '}
                        </BodyShort>
                        <HelpText title="Hva ligger i område?">
                            Viser områdene verge har innsynsrett i.
                            <br />
                            Viser kun tjenesteoppgaver knyttet til NAV.
                        </HelpText>
                    </HStack>
                    <Detail>{verge.tjenesteOppgaver.join(', ')}</Detail>
                </>
            ) : (
                <>
                    <BodyShort weight="semibold" size="small">
                        Omfang
                    </BodyShort>
                    <Detail>{verge.omfang}</Detail>
                </>
            )}
            <Detail textColor="subtle">{verge.embete}</Detail>
            <ValidPeriod from={verge.gyldighetsPeriode?.gyldigFraOgMed} to={verge.gyldighetsPeriode?.gyldigTilOgMed} />
        </InfoElement>
    );
}

function Vergesakstype(props: { vergemal: Verge[] }) {
    const alleVergesakstyper = props.vergemal.map((verge) => verge.vergesakstype);
    const unikeVergessakstyper = Array.from(new Set(alleVergesakstyper)).join(', ');
    return <BodyShort size="small">Vergesakstyper: {unikeVergessakstyper}</BodyShort>;
}

function Vergemal() {
    const { data } = usePersonData();
    const {
        feilendeSystemer,
        person: { vergemal }
    } = data;

    if (vergemal.isEmpty()) {
        return null;
    }

    return (
        <Group
            icon={<ExclamationmarkTriangleFillIcon color="var(--a-icon-warning)" />}
            title="Bruker er under vergemål"
        >
            <Vergesakstype vergemal={vergemal} />
            {vergemal.map((verge, index) => (
                <Verge feilendeSystemer={feilendeSystemer} verge={verge} key={`${verge.ident}-${index}`} />
            ))}
        </Group>
    );
}

export default Vergemal;
