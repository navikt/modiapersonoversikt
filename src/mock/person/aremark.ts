import { Diskresjonskoder, TilrettelagtKommunikasjonsTyper } from '../../konstanter';
import { Kjønn, Person, Relasjonstype, SivilstandTyper } from '../../models/person/person';

export const aremark: Person = {
    fødselsnummer: '10108000398',
    kjønn: Kjønn.Mann,
    geografiskTilknytning: '0118',
    alder: 42,
    navn: {
        sammensatt: 'AREMARK TESTFAMILIEN',
        fornavn: 'AREMARK',
        mellomnavn: '',
        etternavn: 'TESTFAMILIEN',
    },
    tilrettelagtKomunikasjonsListe: [
        {
            behovKode: 'TOHJ',
            beskrivelse: TilrettelagtKommunikasjonsTyper.TOHJ
        }
    ],
    diskresjonskode: Diskresjonskoder.FORTROLIG_ADRESSE,
    statsborgerskap: 'NORSK',
    personstatus: {
        dødsdato: undefined,
        bostatus: undefined
    },
    sivilstand: {
        value: SivilstandTyper.Gift,
        beskrivelse: 'Gift',
        fraOgMed: '2005-12-12'
    },
    familierelasjoner: [{
        harSammeBosted: false,
        rolle: Relasjonstype.Barn,
        tilPerson: {
            navn: {
                sammensatt: '',
                fornavn: 'Siri',
                mellomnavn: null,
                etternavn: 'Aremark'
            },
            alder: 17,
            fødselsnummer: '10117100000',
            personstatus: {}
        }

    }],
    kontaktinformasjon: {
        mobil: {
            sistEndret: '2014-06-21T18:44:39+02:00',
            telefonnummer: '99887766',
            retningsnummer: '47',
            sistEndretAv: 'BRUKER'
        }
    }
};