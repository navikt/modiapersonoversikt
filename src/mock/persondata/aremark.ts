import {
    AdresseBeskyttelse,
    EgenAnsatt,
    ForelderBarnRelasjon,
    ForelderBarnRelasjonRolle,
    FullmaktsRolle,
    Kjonn,
    LocalDate,
    LocalDateTime,
    Person,
    PersonStatus,
    SivilstandType,
    Skifteform
} from '../../app/personside/visittkort-v2/PersondataDomain';
import { harDiskresjonskode } from '../../app/personside/visittkort-v2/visittkort-utils';

export function hentBarnAremark(): ForelderBarnRelasjon[] {
    return [
        {
            ident: '12345678910',
            rolle: ForelderBarnRelasjonRolle.BARN,
            navn: [
                {
                    fornavn: 'BARN1',
                    mellomnavn: null,
                    etternavn: 'BARNESEN'
                }
            ],
            fodselsdato: ['2008-03-15' as LocalDate],
            alder: 13,
            kjonn: [
                {
                    kode: Kjonn.K,
                    beskrivelse: 'Kvinne'
                }
            ],
            adressebeskyttelse: [
                {
                    kode: AdresseBeskyttelse.UGRADERT,
                    beskrivelse: 'UGRADERT'
                }
            ],
            harSammeAdresse: false,
            dodsdato: []
        },
        {
            ident: '12345678911',
            rolle: ForelderBarnRelasjonRolle.BARN,
            navn: [
                {
                    fornavn: 'BARN1',
                    mellomnavn: null,
                    etternavn: 'BARNESEN'
                }
            ],
            fodselsdato: ['2008-03-15' as LocalDate],
            alder: null,
            kjonn: [
                {
                    kode: Kjonn.K,
                    beskrivelse: 'Kvinne'
                }
            ],
            adressebeskyttelse: [],
            harSammeAdresse: false,
            dodsdato: []
        }
    ];
}

export const aremark: Person = {
    fnr: '10108000398',
    navn: [
        {
            fornavn: 'AREMARK',
            mellomnavn: null,
            etternavn: 'TESTFAMILIEN'
        }
    ],
    kjonn: [
        {
            kode: Kjonn.M,
            beskrivelse: 'Mann'
        }
    ],
    fodselsdato: ['1980-10-10' as LocalDate],
    geografiskTilknytning: '0118',
    alder: 41,
    dodsdato: [],
    bostedAdresse: [
        {
            linje1: 'Islandsgate 49',
            linje2: '7693 Hovedøya',
            linje3: 'Norge',
            sistEndret: {
                ident: 'Folkeregisteret',
                tidspunkt: '2010-02-20T10:15:30' as LocalDateTime,
                system: 'FREG'
            },
            gyldighetsPeriode: null
        }
    ],
    kontaktAdresse: [
        {
            linje1: 'Aremarkveien 1',
            linje2: '0320 Aremark',
            linje3: null,
            sistEndret: {
                ident: 'Folkeregisteret',
                tidspunkt: '2020-01-01T10:15:30' as LocalDateTime,
                system: 'FREG'
            },
            gyldighetsPeriode: null
        }
    ],
    oppholdsAdresse: [
        {
            linje1: 'Oppholdsadresse 1',
            linje2: '0000 AREMARK',
            linje3: null,
            sistEndret: null,
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    navEnhet: {
        id: '0219',
        navn: 'NAV Bærum',
        publikumsmottak: [
            {
                besoksadresse: {
                    linje1: 'Adressevei 1',
                    linje2: '0000 AREMARK',
                    linje3: null,
                    sistEndret: null,
                    gyldighetsPeriode: null
                },
                apningstider: [
                    {
                        ukedag: 'Mandag',
                        apningstid: '09.00 - 15.00'
                    },
                    {
                        ukedag: 'Tirsdag',
                        apningstid: '09.00 - 15.00'
                    },
                    {
                        ukedag: 'Onsdag',
                        apningstid: '10.00 - 16.00'
                    }
                ]
            }
        ]
    },
    statsborgerskap: [
        {
            land: {
                kode: 'NOR',
                beskrivelse: 'NORGE'
            },
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    adressebeskyttelse: [
        {
            kode: AdresseBeskyttelse.KODE6,
            beskrivelse: 'Kode 6'
        }
    ],
    sikkerhetstiltak: [
        {
            type: 'TFUS',
            beskrivelse: 'Telefonisk utestengelse',
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    erEgenAnsatt: EgenAnsatt.JA,
    personstatus: [
        {
            kode: PersonStatus.BOSATT,
            beskrivelse: 'Bosatt i Norge'
        }
    ],
    sivilstand: [
        {
            type: {
                kode: SivilstandType.GIFT,
                beskrivelse: 'Gift'
            },
            gyldigFraOgMed: '2005-12-12' as LocalDate,
            sivilstandRelasjon: {
                fnr: '12345555555',
                navn: [
                    {
                        fornavn: 'BAREMARK',
                        mellomnavn: null,
                        etternavn: 'TESTFAMILIEN'
                    }
                ],
                alder: 40,
                adressebeskyttelse: [
                    {
                        kode: AdresseBeskyttelse.UGRADERT,
                        beskrivelse: 'UGRADERT'
                    }
                ],
                harSammeAdresse: true
            }
        }
    ],
    foreldreansvar: hentBarnAremark().map((barn) => ({
        ansvar: 'felles',
        ansvarlig: null,
        ansvarsubject: {
            navn: harDiskresjonskode(barn.adressebeskyttelse) ? null : barn.navn.firstOrNull(),
            ident: barn.ident
        }
    })),
    deltBosted: [
        {
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            },
            adresse: {
                linje1: `Adresseveien 1`,
                linje2: '0000 Aremark',
                linje3: null,
                sistEndret: null,
                gyldighetsPeriode: null
            }
        }
    ],
    dodsbo: [
        {
            adressat: {
                advokatSomAdressat: {
                    kontaktperson: {
                        fornavn: 'Advokat',
                        mellomnavn: null,
                        etternavn: 'Navnesen'
                    },
                    organisasjonsnavn: 'Advokatkontoret Aremark',
                    organisasjonsnummer: '01234567891011'
                },
                organisasjonSomAdressat: null,
                personSomAdressat: null
            },
            adresse: {
                linje1: 'Elgelia 20, 0000 Aremark',
                linje2: null,
                linje3: null,
                sistEndret: null,
                gyldighetsPeriode: null
            },
            registrert: '2010-02-02' as LocalDate,
            skifteform: Skifteform.OFFENTLIG,
            sistEndret: {
                ident: 'Folkeregisteret',
                tidspunkt: '2020-01-01T10:15:30' as LocalDateTime,
                system: 'Folkeregisteret'
            }
        }
    ],
    fullmakt: [
        {
            motpartsPersonident: '123456789',
            motpartsPersonNavn: {
                fornavn: 'Navn',
                mellomnavn: null,
                etternavn: 'Navnesen'
            },
            motpartsRolle: FullmaktsRolle.FULLMEKTIG,
            omrade: [
                {
                    kode: '*',
                    beskrivelse: ''
                }
            ],
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-06-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    vergemal: [
        {
            ident: '21042900076',
            navn: {
                fornavn: 'Simen',
                mellomnavn: null,
                etternavn: 'Solli'
            },
            vergesakstype: 'Voksen',
            omfang: 'Ivareta personens interesser innenfor det personlige og økonomiske området',
            embete: 'Fylkesmannen i Troms og Finnmark',
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-06-01' as LocalDate,
                gyldigTilOgMed: '2022-05-10' as LocalDate
            }
        },
        {
            ident: '123456799',
            navn: {
                fornavn: 'Truls',
                mellomnavn: null,
                etternavn: 'Tøffel'
            },
            vergesakstype: 'Fremtidsfullmakt',
            omfang: 'Ivareta personens interesser innenfor det økonomiske området',
            embete: 'Fylkesmannen i Troms og Finnmark',
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-10' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    tilrettelagtKommunikasjon: {
        talesprak: [
            {
                kode: 'NO',
                beskrivelse: 'Norsk'
            }
        ],
        tegnsprak: [
            {
                kode: 'NO',
                beskrivelse: 'Norsk'
            }
        ]
    },
    telefonnummer: [
        {
            retningsnummer: {
                kode: '+47',
                beskrivelse: 'Norge'
            },
            identifikator: '99009900',
            sistEndret: {
                ident: '11223344',
                tidspunkt: '2018-06-14T00:00:00' as LocalDateTime,
                system: 'BD06'
            },
            prioritet: -1
        }
    ],
    kontaktOgReservasjon: {
        personident: '10108000398',
        reservasjon: 'true',
        epostadresse: {
            value: 'epost@nav.no',
            sistOppdatert: null,
            sistVerifisert: '2013-01-01' as LocalDate
        },
        mobiltelefonnummer: {
            value: '12345678',
            sistOppdatert: '2021-11-02' as LocalDate,
            sistVerifisert: '2021-11-02' as LocalDate
        }
    },
    bankkonto: {
        kontonummer: '12345678900',
        banknavn: 'DNB ASA',
        sistEndret: {
            ident: '1010800 BD03',
            tidspunkt: '2006-03-15T00:00:00' as LocalDateTime,
            system: ''
        },
        bankkode: null,
        swift: 'DNBANOKKXXX',
        landkode: null,
        adresse: {
            linje1: 'Bankveien 1,',
            linje2: '0357 Bankestad',
            linje3: null,
            sistEndret: null,
            gyldighetsPeriode: null
        },
        valuta: {
            kode: 'NOK',
            beskrivelse: 'Norske kroner'
        }
    },
    forelderBarnRelasjon: hentBarnAremark()
};
