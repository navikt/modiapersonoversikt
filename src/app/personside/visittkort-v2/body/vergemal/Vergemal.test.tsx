import * as renderer from 'react-test-renderer';
import { aremark } from '../../../../../mock/persondata/aremark';
import TestProvider from '../../../../../test/Testprovider';
import Vergemal from './Vergemal';

test('viser vergemÃ¥l', () => {
    const visittkortbody = renderer.create(
        <TestProvider>
            <Vergemal feilendeSystemer={[]} vergemal={aremark.vergemal} />
        </TestProvider>
    );

    expect(visittkortbody.toJSON()).toMatchSnapshot();
});
