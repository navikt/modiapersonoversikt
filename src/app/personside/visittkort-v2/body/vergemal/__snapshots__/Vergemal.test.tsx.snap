// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`viser vergemål 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  position: absolute;
  left: -3.125rem;
  width: 3.125rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 > svg {
  height: 1.5rem;
  width: auto;
}

.c3 {
  margin: 10px 0 20px 0;
}

.c3:last-child {
  margin: 10px 0 0 0;
}

.c5 {
  color: #645f5a;
  margin-top: 3px;
}

.c5 .typo-etikett-liten {
  line-height: 1rem;
}

.c2 {
  margin-bottom: 1rem;
}

.c4 {
  margin-bottom: 5px;
}

<section>
  <div
    className="c0"
  >
    <div
      className="c1"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="50"
          cy="50"
          fill="#FFB249"
          r="50"
        />
        <line
          stroke="#3E3832"
          strokeLinecap="round"
          strokeWidth="4"
          x1="50"
          x2="50"
          y1="25"
          y2="57"
        />
        <circle
          cx="50"
          cy="72"
          fill="#3E3832"
          r="4.5"
        />
      </svg>
    </div>
    <h3
      className="typo-undertittel"
    >
      Bruker er under vergemål
    </h3>
  </div>
  <div
    className="c2"
  />
  <p
    className="typo-normal"
  >
    Vergesakstyper: 
    Voksen, Fremtidsfullmakt
  </p>
  <div
    className="c3"
  >
    <div
      className="c0"
    >
      
      <h4
        className="typo-element"
      >
        Verge
      </h4>
    </div>
    <div
      className="c4"
    >
      <p
        className="typo-normal"
      >
        Simen Solli
      </p>
      <p
        className="typo-normal"
      >
        21042900076
      </p>
    </div>
    <p
      className="typo-element"
    >
      Omfang
    </p>
    <p
      className="typo-normal"
    >
      Ivareta personens interesser innenfor det personlige og økonomiske området
    </p>
    <div
      className="c5"
    >
      <p
        className="typo-undertekst"
      >
        Fylkesmannen i Troms og Finnmark
      </p>
    </div>
    <div
      className="c5"
    >
      <p
        className="typo-undertekst"
      >
        Gyldig: 
        01.06.2021 – 10.05.2022
      </p>
    </div>
  </div>
  <div
    className="c3"
  >
    <div
      className="c0"
    >
      
      <h4
        className="typo-element"
      >
        Verge
      </h4>
    </div>
    <div
      className="c4"
    >
      <p
        className="typo-normal"
      >
        Truls Tøffel
      </p>
      <p
        className="typo-normal"
      >
        123456799
      </p>
    </div>
    <p
      className="typo-element"
    >
      Område
       
      <div
        className="hjelpetekst"
      >
        <button
          aria-controls="Helt tilfeldig ID"
          aria-expanded={false}
          aria-haspopup="dialog"
          className="hjelpetekst__apneknapp"
          onClick={[Function]}
          title="Hjelp"
          type="button"
        >
          <svg
            className="hjelpetekst__ikon"
            focusable="false"
            height={32}
            kind="help-circle"
            viewBox="0 0 18.2 18.2"
            width={32}
          >
            <path
              d="M9.1 0C4.1 0 0 4.1 0 9.1s4.1 9.1 9.1 9.1 9.1-4.1 9.1-9.1S14.1 0 9.1 0zm0 17.2C4.6 17.2 1 13.6 1 9.1S4.6 1 9.1 1s8.1 3.6 8.1 8.1-3.6 8.1-8.1 8.1z"
            />
            <circle
              cx="9.1"
              cy="13.8"
              r=".9"
            />
            <path
              d="M9.1 11.5c-.3 0-.5-.2-.5-.5V8.6c0-.3.2-.5.5-.5 1 0 1.9-.8 1.9-1.9s-.8-1.9-1.9-1.9c-1 0-1.9.8-1.9 1.9 0 .3-.2.5-.5.5s-.5-.2-.5-.4c0-1.6 1.3-2.9 2.9-2.9S12 4.7 12 6.3c0 1.4-1 2.6-2.4 2.8V11c0 .3-.2.5-.5.5z"
            />
          </svg>
          <span
            className="sr-only"
          >
            Hjelp
          </span>
        </button>
        <div
          className="popover popover--controlled popover--hidden popover--over"
          id="Helt tilfeldig ID"
          onClick={[Function]}
          role="tooltip"
          style={
            {
              "transform": "translate3d(0px, 0px, 0)",
            }
          }
          tabIndex={0}
        >
          <div
            className="popover__content-inner"
          >
            <div
              className="hjelpetekst__innhold"
            >
              Viser områdene verge har innsynsrett i.
              <br />
              Viser kun tjenesteoppgaver knyttet til NAV.
            </div>
          </div>
          <span
            className="popover__pil"
            style={
              {
                "left": undefined,
              }
            }
          />
        </div>
      </div>
    </p>
    <p
      className="typo-normal"
    >
      Arbeid, Familie, Hjelpemidler, Sosiale Tjenester
    </p>
    <div
      className="c5"
    >
      <p
        className="typo-undertekst"
      >
        Fylkesmannen i Troms og Finnmark
      </p>
    </div>
    <div
      className="c5"
    >
      <p
        className="typo-undertekst"
      >
        Gyldig: 
        10.02.2021 – 01.05.2022
      </p>
    </div>
  </div>
</section>
`;
